test/minishift/test_suite.go

```diff
@@ -0,0 +1,161 @@
+package minishift
+
+import (
+	"context"
+	"crypto/tls"
+	"github.com/fabric8-services/fabric8-common/log"
+	"github.com/fabric8-services/fabric8-tenant/auth"
+	"github.com/fabric8-services/fabric8-tenant/cluster"
+	"github.com/fabric8-services/fabric8-tenant/configuration"
+	"github.com/fabric8-services/fabric8-tenant/environment"
+	"github.com/fabric8-services/fabric8-tenant/openshift"
+	"github.com/fabric8-services/fabric8-tenant/test"
+	"github.com/fabric8-services/fabric8-tenant/test/gormsupport"
+	"github.com/fabric8-services/fabric8-tenant/test/resource"
+	"github.com/fabric8-services/fabric8-tenant/test/stub"
+	"github.com/satori/go.uuid"
+	"github.com/stretchr/testify/assert"
+	"net/http"
+	"os"
+	"sync"
+	"testing"
+	"time"
+)
+
+// TestSuite is a base for tests using Minishift and gorm db
+type TestSuite struct {
+	gormsupport.DBTestSuite
+	ClusterService      *stub.ClusterService
+	AuthService         *stub.AuthService
+	Config              *configuration.Data
+	toReset             func()
+	minishiftUrl        string
+	minishiftAdminName  string
+	minishiftAdminToken string
+	minishiftUserName   string
+	minishiftUserToken  string
+}
+
+func (s *TestSuite) SetupTest() {
+	resource.Require(s.T(), resource.Database, "F8_MINISHIFT")
+	s.minishiftUrl = os.Getenv("F8_MINISHIFT_URL")
+	s.minishiftAdminName = os.Getenv("F8_MINISHIFT_ADMIN_NAME")
+	s.minishiftAdminToken = os.Getenv("F8_MINISHIFT_ADMIN_TOKEN")
+	s.minishiftUserName = os.Getenv("F8_MINISHIFT_USER_NAME")
+	s.minishiftUserToken = os.Getenv("F8_MINISHIFT_USER_TOKEN")
+	s.DBTestSuite.SetupTest()
+	s.Config, s.toReset = prepareConfig(s.T())
+
+	log.InitializeLogger(s.Config.IsLogJSON(), s.Config.GetLogLevel())
+
+	s.ClusterService = &stub.ClusterService{
```

Comment:
this could be initialized in the `SetupSuite` function, once for all

Reply:
I want to keep it consistent - use the same approach for both ClusterServices and AuthService